### PR TITLE
#609 fix: auth vulnerability

### PR DESCRIPTION
**Pull request by vinitparekh17**
### This PR resolves the auth vulnerability

1. merges flag value to the global config object in order to prioritize `-requirepass` flag
2. shows warning if dice starts without password
3. uses `config.DiceConfig.Auth.UserName` as a default user

### CODE REVIEW

**Comment by lucifercr07** [comment]
@vinitparekh17 can you create a tracker issue for this for tracking purposes, with details around the bug.

**Comment by vinitparekh17** [comment]
@lucifercr07 Is there any particular format for issue or I will do my way?

**Comment by vinitparekh17** [comment]
This PR resolves issue #609

File: config/config.go [hunk 1]
```diff
@@ -237,6 +240,12 @@ func setUpViperConfig(configFilePath string) {
 	}
 
 	// override default configurations with command line flags
+	mergeFlagsWithConfig()
+
+	log.Info("configurations loaded successfully.")
```
Comment (JyotinderSingh) [inline config/config.go:245]: not needed.
```suggestion
```

**Review by JyotinderSingh — APPROVED** [review]
Thanks for the fix @vinitparekh17! The changes look good.

**Pull request merged**